workspaces/core/src/modal/Modal.tsx

```diff
@@ -8,19 +14,39 @@ type WithChildren<T = {}> = T & { children?: ReactNode };
 type ModalRootProps = {
   open: boolean;
   onClose: () => void;
+  initialFocus?: React.MutableRefObject<HTMLElement | null>;
 };
 
 type ModalComponentProps = (
   props: WithChildren<ModalRootProps>
 ) => React.ReactElement | null;
 
-const ModalRoot: ModalComponentProps = ({ open, onClose, children }) => (
-  <Transition appear show={open} as={React.Fragment}>
-    <Dialog as="div" className="relative z-10" onClose={onClose}>
-      {children}
-    </Dialog>
-  </Transition>
-);
+const ModalRoot: ModalComponentProps = ({
+  open,
+  onClose,
+  children,
+  initialFocus,
+}) => {
+  const defFocus = useRef(null);
+  const [focusElRef, setFocusElRef] =
+    useState<MutableRefObject<HTMLElement | null>>();
+
+  useEffect(() => {
+    initialFocus ? setFocusElRef(initialFocus) : setFocusElRef(defFocus);
+  }, [initialFocus]);
```

Comment:
I see 2 options here

1) 

```
useEffect(() => {
  setFocusElRef(initialFocus ? initialFocus : defFocus);
}, [initialFocus, defFocus]);
```

or

2) A simple ternary condition is enough (imho)
```
const defFocus = useRef<HTMLElement | null>(null);
const focusElRef = initialFocus ? initialFocus : defFocus;

return (
  <Transition appear show={open}>
    <Dialog
        ....
  </Transition>
);
```

Reply:
Fixed with the first option. Unfortunately, I had to implement with the help of useEffect for proper work on the brand side with SSR.